routers/web/repo/repo.go

```diff
@@ -51,7 +51,7 @@ func MustBeNotEmpty(ctx *context.Context) {
 
 // MustBeEditable check that repo can be edited
 func MustBeEditable(ctx *context.Context) {
-	if !ctx.Repo.Repository.CanEnableEditor() || ctx.Repo.IsViewCommit {
+	if !ctx.Repo.Repository.CanEnableEditor() {
```

Comment:
`ctx.Repo.IsViewCommit` is not needed because it doesn't belong to the "context repo", and the ref type is checked by the following middlewares like "RefTypeByBranch"